tests/Meziantou.Analyzer.Test/Rules/AvoidComparisonWithBoolConstantAnalyzerTests.cs

```diff
@@ -163,6 +163,34 @@ await CreateProjectBuilder()
                   .ValidateAsync();
         }
 
+        [Theory]
+        [InlineData("dynamicValue == true")]
+        [InlineData("true == AsDynamic().MaybeBoolean")]
+        [InlineData("((dynamic)this.TrulyBoolean) == true")]
+        public async Task ComparingDynamicVariableWithBoolLiteral_NoDiagnosticReported(string expression)
+        {
+            var originalCode = $@"
+class TestClass
+{{
+    public bool? MaybeBoolean {{ get; set; }}
+    public bool  TrulyBoolean {{ get; set; }}
+
+    public dynamic AsDynamic() {{ return this; }}
+
+    void Test()
+    {{
+        dynamic dynamicValue = true;
+        if ({expression})
+        {{
+        }}
+    }}
+}}";
+            await CreateProjectBuilder()
+                  .WithSourceCode(originalCode)
+                  .AddDynamicTypeSupport()
```

Comment:
I made the support for `dynamic` type optional.
@meziantou, if you want I'll make it a standard thing.

Reply:
I think it should be standard